plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java

```diff
@@ -603,11 +607,41 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
                 handle.getRemoteTableName(),
                 handle.getFilter(),
                 newDomain,
+                handle.getProjectedColumns(),
                 handle.getLimit());
 
         return Optional.of(new ConstraintApplicationResult<>(handle, remainingFilter, false));
     }
 
+    @Override
+    public Optional<ProjectionApplicationResult<ConnectorTableHandle>> applyProjection(
+            ConnectorSession session,
+            ConnectorTableHandle handle,
+            List<ConnectorExpression> projections,
+            Map<String, ColumnHandle> assignments)
+    {
+        MongoTableHandle mongoTableHandle = (MongoTableHandle) handle;
+
+        Set<ColumnHandle> newColumns = ImmutableSet.copyOf(assignments.values());
+
+        if (mongoTableHandle.getProjectedColumns().isPresent() && newColumns.equals(mongoTableHandle.getProjectedColumns().get())) {
```

Comment:
`mongoTableHandle.getProjectedColumns().isPresent() &&`
can this condition be removed?

Reply:
No. It's needed.